src/components/TrailForm.js

```diff
@@ -160,8 +178,45 @@ class TrailForm extends React.Component {
 
             </List>
 
+            <Button
+              fullWidth
+              style={{ marginTop: 10 }}
+              color="secondary"
+              variant="raised"
+              onClick={() => { this.toggleConfirmation(true)}}
+            > Delete Trail
+            </Button>
+
           </CardContent>
         </Card>
+
+        <Dialog open={dialogOpen} >
+
+          <DialogTitle>Confirm Trail Delete</DialogTitle>
+
+          <DialogContent>
+            <DialogContentText id="alert-dialog-description">
+               You are about to delete a trail. Deleting a trail will remove
+               the trail polygon features and dissascociate the hydrants currently assigned.
+            </DialogContentText>
+          </DialogContent>
+
+          <DialogActions>
+            <Button
+              color="primary"
+              onClick={() => { trailDeleted(editableTrail.get('id'))}}
```

Comment:
Confirming delete should also close the dialog window, is that happening somewhere else?

Reply:
Delete closes the dialog since it's currently a part of trail form which only renders if the trail is editable. Since editableTrail is set to null after trail deletion, the dialog goes away along w/ the trail form.